### ISSUE TITLE
Roll out comprehensive documentation for ZIO HTTP

### ISSUE THREAD

**Issue by jdegoes** [zio/zio-http#2198]
The documentation can be patterned on [http4k](https://www.http4k.org/documentation/), and provide a variety of sections, including:

 - Introduction
 - Concepts
   - Routing
   - Request Handling
   - Server
   - Client
   - Middleware
   - Endpoint
 - Tutorials
   - Your First ZIO HTTP App
   - Deploying a ZIO HTTP App 
   - Testing Your ZIO HTTP App
 - How-to-guides
   - (This should be based off the list of topics at http4k, but should also include ZIO HTTP use cases around Endpoint, middleware, etc.)
 - Reference
   - API Docs
   - Server backend
   - Websockets
   - JSON Handling
   - Metrics
   - Request Logging
   - (etc.)
 - Performance
 - FAQ

The documentation must be built with existing tooling, include examples that are checked for compile-ability with `mdoc`, and must be copyedited for correct spelling and grammar (possibly with the help of AI tooling).

The overall volume of content would be similar to, but less than, http4k's documentation website.

**Comment by harshbisle** [comment]
`/attempt #2198`

**Comment by harshbisle** [comment]
Yes, I will be finishing the zio-cli doc this week and eventually hop on to this issue.

**Comment by daveads** [comment]
This would require me to dig in and understand how zio-http works 

@jdegoes  is someone working on this already ?  and can i use [mdbook](https://rust-lang.github.io/mdBook/) for this ?

**Comment by daveads** [comment]
@feliciien

**Comment by feliciien** [comment]
@daveads hello? can we work together?

**Comment by daveads** [comment]
@feliciien  let's talk on discord  username: **daveads#6337**

**Comment by harshbisle** [comment]
`/attempt #2198` Hey @daveads and @feliciien, can I join? What's your progress, BTW?

**Comment by feliciien** [comment]
i send you  friend request

**Comment by wright-eric** [comment]
/attempt #2198

<details id="algora-options">
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clhrg2sx60000mk0fnta1xclv/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by varshith257** [comment]
/attempt #2198

<div id="algora" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clhrg2sx60000mk0fnta1xclv/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by varshith257** [comment]
@khajavi @jdegoes Is it ready to work?

**Comment by varshith257** [comment]
@jdegoes Need your guidance here

**Comment by varshith257** [comment]
@khajavi Let's wait for the response of @daveads for a while and others or I can proceed further to make progress on this if it doesn't get any response on this issue from daveads.

Seems others dropped from long ago with the response of @daveads making progress from a year.

**Comment by daveads** [comment]
> @khajavi Let's wait for the response of @daveads for a while and others or I can proceed further to make progress on this if it doesn't get any response on this issue from daveads.
> 
> Seems others dropped from long ago with the response of @daveads making progress from a year.

Decided to break down the PR into smaller pull request so it would be easy for reviewers to review it and then point to them when am finally done... something in that order

**Comment by varshith257** [comment]
cc: @khajavi 
I am working alone and will submit PR soon :)

**Comment by promisingcoder** [comment]
/attempt #2198 

<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clhrg2sx60000mk0fnta1xclv/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by varshith257** [comment]
Hey @promisingcoder, I have done some cool progress on it and waiting for maintainers reviews. It could be duplicate of efforts again if you get started with this. So I think you can have a look at other open issues :)

**Comment by Thund3rHawk** [comment]
@jdegoes hi! Is this issue still open?

**Comment by hoklims** [comment]
/attempt #2198

<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clhrg2sx60000mk0fnta1xclv/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by zelosleone** [comment]
/attempt #2198

<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clhrg2sx60000mk0fnta1xclv/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by toluwagbemiga** [comment]
/attempt #2198

<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clhrg2sx60000mk0fnta1xclv/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by toluwagbemiga** [comment]
hello, is the issue still open?

**Comment by albassort** [comment]
@jdegoes it seems this Issue should be closed. Almost all of this has been merged into the contemporary page. I don't even know what I should be working on here.

**Comment by Brijeshthummar02** [comment]
@987Nabil is this issue still open?

**Comment by albassort** [comment]
> [@987Nabil](https://github.com/987Nabil) is this issue still open?

I wouldn't bother all of the stuff requested is implemented, the issue is vestigial

**Comment by Govindggupta** [comment]
Hello folks,

I recently came across [issue #2198](https://github.com/zio/zio-http/issues/2198). I noticed that most of the previous attempts on this issue seem quite old, so I wanted to confirm whether this issue is still open and if work on it is still needed.

Also, the reference link to [http4k's documentation](https://www.http4k.org/documentation/) provided in the issue description shows       404 error for me. Could someone please share another reference link so I can review the recommended structure and begin working on the documentation?

**Comment by khajavi** [comment]
Hello @Govindggupta 

Yeah! This issue is outdated. However, contributions to the ZIO HTTP documentation are still welcome. Ping me if you'd like to contribute!

**Comment by Govindggupta** [comment]
Sure @khajavi I would love to contribute to the ZIO HTTP documentation. 
Can you tell me how can i contribute to it ?
